internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -496,6 +509,32 @@ func (r *BMACReconciler) reconcileAgentInventory(log logrus.FieldLogger, bmh *bm
 
 }
 
+// Ask BMH to reboot the host if the agent is unbound after installation
+//
+// By re-attaching the BMH and clearing the Image field on it, BMAC will clear
+// the Image data to force the boot from ISO
+//
+func (r *BMACReconciler) reconcileUnboundAgent(log logrus.FieldLogger, bmh *bmh_v1alpha1.BareMetalHost, agent *aiv1beta1.Agent) reconcileResult {
+	log.Debugf("Started Unbound Agent reconcile for agent %s/%s and bmh %s/%s", agent.Namespace, agent.Name, bmh.Namespace, bmh.Name)
+
+	// proceed with the reconcile only when the agent ask for user action following unbinding
+	boundCondition := conditionsv1.FindStatusCondition(agent.Status.Conditions, aiv1beta1.BoundCondition)
+	if boundCondition == nil || boundCondition.Reason != aiv1beta1.UnbindingPendingUserActionReason {
+		return reconcileComplete{}
```

Comment:
It'd be useful to have a debug line here mentioning that we are skipping the reconcile for unbound agents

Reply:
It was not called by design because in most case the call will be skipped. In Debug mode the log will be littered with these non useful logs. The case where we do find unbound case is clearly documented